scalaz-test/src/main/scala/ztest/IOSuite.scala

```diff
@@ -0,0 +1,67 @@
+package scalaz.test
+
+import scala._, scala.Predef._
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+
+import scalaz.data.IList.cons
+import scalaz.Void
+import scalaz.data._
+import scalaz.Scalaz._
+
+import scalaz.effect.{ IO, IORef }
+
+abstract class IOSuite[E] extends Suite {
+  def printError(e: E): TestResult
+  def doTests[G[_]: Monad](test: Harness[IO[E, ?], G]): G[Unit]
```

Comment:
Can't help this one:

![G[Unit]](https://images-na.ssl-images-amazon.com/images/I/81YhjssTn3L._SX522_.jpg)